### ISSUE TITLE
Improve contents visibility for transparent panels

### ISSUE THREAD

**Issue by psifidotos** [psifidotos/Latte-Dock#905]
to improve contrast with the underlying background
Latte now calculates the luminosity of the area
underneath the panel and based on that chooses the
best color to create contrast. The colors are chosen
only between textColor and backgroundColor.

**Comment by psifidotos** [comment]
the following screenshots are from the same Latte panel when is in different Activities and edges... :) enjoy...

![default](https://user-images.githubusercontent.com/2216474/37102344-53280322-2230-11e8-9dfd-304e358f24a1.png)

![default](https://user-images.githubusercontent.com/2216474/37102367-5fcc98cc-2230-11e8-9d4d-5132517f7f09.png)

![default](https://user-images.githubusercontent.com/2216474/37102413-72dd481c-2230-11e8-9987-7557070ef7c8.png)

**Comment by michaldybczak** [comment]
Thanks, that looks cool. Will definitely try it out!

At the moment I'm very happy with latte settings possibilities because I can set everything I need: transparent, non-transparent, semi-transparent, none-transparent on maximized state, own defined transparency level, colors or whatever. This is truly the best dock there is! :D And as a happy user, I keep using it quietly (there are no issues, all works great) :).

**Comment by michaldybczak** [comment]
OK, I looked closer on the new feature.
Here is my current setting (**turned off** "improve content visibility when panel is transparent):
![screenshot_20180308_070355](https://user-images.githubusercontent.com/10132958/37135850-684de6ee-229f-11e8-9c40-3415c8444964.jpg)

Here is my current setting (**turned on** "improve content visibility when panel is transparent):
![screenshot_20180308_070415](https://user-images.githubusercontent.com/10132958/37135861-79dd2e4c-229f-11e8-857d-c2df8058531d.jpg)

If I turn off the background, result with this feature is:
![screenshot_20180308_070903](https://user-images.githubusercontent.com/10132958/37135912-adc7bf74-229f-11e8-9390-d330cc0136d5.jpg)

Changed wallpaper for dark and it looks like that:
![screenshot_20180308_071149](https://user-images.githubusercontent.com/10132958/37136003-fd838b92-229f-11e8-857a-23d90d8fd69b.jpg)

With light background wallpaper:
![screenshot_20180308_071430](https://user-images.githubusercontent.com/10132958/37136091-67731040-22a0-11e8-902c-0a5c5d112a03.jpg)

So basically at the moment the new setting is not adjusting itself intelligently but... there is a case where it adds value, mainly: if I want to have dark background (so I set dark theme) somewhere (like on a dock) but still light wallpaper with transparent panel and fonts dark (instead white ones that come with dark themes). That's the case showed in the last screen.

Or maybe it doesn't work correctly for me? In your examples the new feature was only changing font color from white to dark/black (with dark theme, dark panel background). Maybe that's result of my combination of settings?

**Comment by psifidotos** [comment]
Doesnt loik like it works for you...

1. Are you using latest master?
2. These backgrounds are static files or slideshow?

**Comment by michaldybczak** [comment]
Backgrounds are static and I updated latte-git from AUR 13 hours ago.

P.S. By the way, I will probably send pull request (when I find time) to polish translation in the feature, because there are some mistakes and missing pieces. For example I have such sentence (translating literally from polish one):

"Watercourse of hidden panel for maximized windows"

I suspect that there is a misspelling and "watercourse' was meant to be "a shadow" and the whole sentence should be something like that:
"Hide panel shadow for maximized windows".
Or maybe should I open a new issue, to keep this one clean?
